bazel: add roachprod-bench dev command

herkolategan opened the pull request:
Add command roachprod-bench to dev that will build all test binaries for a given package. On a successful build a process will be started that copies and runs the benchmarks present in the binaries across a given roachprod cluster. Benchmark results and analytics are collected and stored in a report for further analysis.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/90696)
<!-- Reviewable:end -->

herkolategan committed:
bazel: add roachprod-bench dev command

Add command roachprod-bench to dev that will build all test binaries
for a given package. On a successful build a process will be started
that copies and runs the benchmarks present in the binaries across
a given roachprod cluster. Benchmark results and analytics are
collected and stored in a report for further analysis.

Release note: None
